From: Patrick Brunschwig <patrick@enigmail.net>
Date: Mon, 2 Apr 2018 19:03:07 +0200
Subject: ensure that there is no exception if homedir doesn't exist

(cherry picked from commit a9877ae3fdda4bafb6de67cf79d1ddd39cd4b2f6)
---
 package/gpgAgent.jsm | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/package/gpgAgent.jsm b/package/gpgAgent.jsm
index f08e65f..e1548a8 100644
--- a/package/gpgAgent.jsm
+++ b/package/gpgAgent.jsm
@@ -733,7 +733,9 @@ var EnigmailGpgAgent = {
     let homeDirObj = Components.classes[NS_LOCAL_FILE_CONTRACTID].createInstance(Ci.nsIFile);
     EnigmailFiles.initPath(homeDirObj, homeDir);
 
-    homeDirObj.normalize(); // resolve symlinks etc.
+    if (homeDirObj.exists()) {
+      homeDirObj.normalize(); // resolve symlinks etc.
+    }
 
     let dirType = EnigmailFiles.ensureWritableDirectory(homeDirObj, 0x1C0); // 0700
     let errMsg = "";
